core/broadleaf-profile-web/src/main/java/org/broadleafcommerce/profile/web/site/security/SessionFixationProtectionFilter.java

```diff
@@ -1,120 +0,0 @@
-/*
- * #%L
- * BroadleafCommerce Profile Web
- * %%
- * Copyright (C) 2009 - 2016 Broadleaf Commerce
- * %%
- * Licensed under the Broadleaf Fair Use License Agreement, Version 1.0
- * (the "Fair Use License" located  at http://license.broadleafcommerce.org/fair_use_license-1.0.txt)
- * unless the restrictions on use therein are violated and require payment to Broadleaf in which case
- * the Broadleaf End User License Agreement (EULA), Version 1.1
- * (the "Commercial License" located at http://license.broadleafcommerce.org/commercial_license-1.1.txt)
- * shall apply.
- * 
- * Alternatively, the Commercial License may be replaced with a mutually agreed upon license (the "Custom License")
- * between you and Broadleaf Commerce. You may not use this file except in compliance with the applicable license.
- * #L%
- */
-package org.broadleafcommerce.profile.web.site.security;
-
-import org.apache.commons.lang.StringUtils;
-import org.apache.commons.logging.Log;
-import org.apache.commons.logging.LogFactory;
-import org.broadleafcommerce.common.encryption.EncryptionModule;
-import org.broadleafcommerce.common.security.RandomGenerator;
-import org.broadleafcommerce.common.security.util.CookieUtils;
-import org.broadleafcommerce.common.util.BLCRequestUtils;
-import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.beans.factory.annotation.Qualifier;
-import org.springframework.security.core.context.SecurityContextHolder;
-import org.springframework.stereotype.Component;
-import org.springframework.web.context.request.ServletWebRequest;
-import org.springframework.web.filter.GenericFilterBean;
-
-import java.io.IOException;
-import java.security.NoSuchAlgorithmException;
-
-import javax.servlet.FilterChain;
-import javax.servlet.ServletException;
-import javax.servlet.ServletRequest;
-import javax.servlet.ServletResponse;
-import javax.servlet.http.HttpServletRequest;
-import javax.servlet.http.HttpServletResponse;
-import javax.servlet.http.HttpSession;
-
-/**
- * Filter used to protected against session fixation attacks while still keeping the same session id on both
- * http and https protocols. Uses a secondary, https cookie that must be present on every https request for a
- * given session after the first request. If it's not present and equal to what we expect, we will redirect the
- * user to "/" and remove his session cookie.
- *
- * @author Andre Azzolini (apazzolini)
- */
-@Component("blSessionFixationProtectionFilter")
-public class SessionFixationProtectionFilter extends GenericFilterBean {
```

Comment:
Bring this class back and add @Deprecated to it with javadocs explaining that it's be replaced by Spring's default implementation